lib/charms/opensearch/v0/opensearch_relation_peer_cluster.py

```diff
@@ -1528,3 +1657,32 @@ def get_cluster_first_data_node(self) -> str | None:
             return None
 
         return self.peer_cm.rel_data_from_str(peer_cluster_data).first_data_node
+
+    def clean_is_candidate_failover(self) -> None:
+        """Remove is_candidate_failover_orchestrator key from all relations"""
+        for rel in self.model.relations[self.relation_name]:
+            self.delete_from_rel(key="is_candidate_failover_orchestrator", rel_id=rel.id)
+
+    def _clean_main_orchestrator_is_requirer_status(self, departing_relation: Relation) -> None:
+        """Block the charm if main orchestrator is a requirer"""
+        if (
+            not self.charm.unit.is_leader()
+            or not (deployment_desc := self.peer_cm.deployment_desc())
+            or deployment_desc.typ != DeploymentType.MAIN_ORCHESTRATOR
+        ):
+            return
+
+        peer_cluster_requirer_relations = [
+            rel
+            for rel in self.model.relations[PeerClusterRelationName]
+            if rel.id != departing_relation.id
+        ]
+        # clean the status if it is set
+        if (
+            not peer_cluster_requirer_relations
+            and self.charm.app.status.message == PClusterMainIsRequirer
+        ):
+            self.charm.status.set(
+                ActiveStatus(),
+                app=True,
+            )
```

Comment:
we shouldn't set directly an ActiveStatus, which is too far reaching, rather `status.clear`